### DISCUSSION TITLE
How can I slice a column data while column resizing?

### DISCUSSION THREAD

**Discussion by baymac** [TanStack/table#2413] [General] [answered]
The behaviour shown in column resizing [example](https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/column-resizing) is that when column is resized to a width lower than the width of column data, the value overflows out of the table if it is a single word or it is pushed into a new line if multiple words. E.g.:
<img width="578" alt="Screenshot 2020-06-06 at 7 24 59 PM" src="https://user-images.githubusercontent.com/23079344/83945958-882e0980-a82b-11ea-9edc-eb4bbfc05137.png">

I would like to slice the column data when the column is resized and column data minimum width exceeds the resized column width. My expectations is sth like:

```javascript
const offset = 3
if(dataMinWidth < columnWidth) {
  data = `${data.slice(0, columnWidth-offset)}...`
}
```

Is there a way to pass a function to format data in each column based on a given function (as above)?

**Comment by iaremarkus** [comment, marked as answer]
I don't think a function would be necessary. This can be achieved with good 'ol CSS:

```css
overflow: hidden;
text-overflow: ellipsis;
white-space: nowrap;
```

You could optionally add a title tag to the `td` with the full string so that it can be viewed by a long-hover.

**Reply by baymac** [reply]
Thanks a lot :)
